www/pages/stop/stop-controller.js

```diff
@@ -207,11 +229,14 @@ angular.module('pvta.controllers').controller('StopController', function ($scope
 
   // **Sets** whether a route's
   // departures have been expanded on the page
-  $scope.toggleGroup = function (group) {
-    if ($scope.isGroupShown(group)) {
+  $scope.toggleGroup = function (routeId, numberOfDepartures) {
+    if (numberOfDepartures === 1 && !_.contains($scope.onlyOneDeparture, routeId)) {
+      $scope.onlyOneDeparture.push(routeId);
+    }
```

Comment:
I'm a noob here, but handling "no upcoming departures" seems like view logic to me